validator/coordinator/src/test/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinatorTest.java

```diff
@@ -13,54 +13,115 @@
+import static tech.pegasys.artemis.datastructures.blocks.BeaconBlockBodyLists.createAttestations;
 
 import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.util.MockStartValidatorKeyPairFactory;
 import tech.pegasys.artemis.statetransition.AttestationAggregator;
 import tech.pegasys.artemis.statetransition.BeaconChainUtil;
 import tech.pegasys.artemis.statetransition.BlockAttestationsPool;
 import tech.pegasys.artemis.statetransition.events.BroadcastAttestationEvent;
 import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.storage.events.SlotEvent;
+import tech.pegasys.artemis.util.Waiter;
+import tech.pegasys.artemis.util.bls.BLSKeyPair;
 import tech.pegasys.artemis.util.config.ArtemisConfiguration;
 
 public class ValidatorCoordinatorTest {
 
+  private BlockAttestationsPool blockAttestationsPool;
+  private AttestationAggregator attestationAggregator;
+  private EventBus eventBus;
+  private ChainStorageClient storageClient;
+  private ArtemisConfiguration config;
   private ValidatorCoordinator vc;
 
-  @Test
-  void noValidators_noAttestationProduced() throws Exception {
-    int numValidators = 0;
+  private final int numValidators = 12;
 
-    ArtemisConfiguration config = mock(ArtemisConfiguration.class);
+  @BeforeEach
+  void setup() {
+    config = mock(ArtemisConfiguration.class);
     doReturn(0).when(config).getNaughtinessPercentage();
     doReturn(numValidators).when(config).getNumValidators();
     doReturn(null).when(config).getValidatorsKeyFile();
     doReturn(0).when(config).getInteropOwnedValidatorStartIndex();
     doReturn(numValidators).when(config).getInteropOwnedValidatorCount();
 
-    EventBus eventBus = spy(new EventBus());
-    AttestationAggregator attestationAggregator = mock(AttestationAggregator.class);
-    BlockAttestationsPool blockAttestationsPool = mock(BlockAttestationsPool.class);
-    final ChainStorageClient storageClient = new ChainStorageClient(eventBus);
-    final BeaconChainUtil chainUtil = BeaconChainUtil.create(numValidators, storageClient);
+    attestationAggregator = mock(AttestationAggregator.class);
+    blockAttestationsPool = mock(BlockAttestationsPool.class);
+    doReturn(createAttestations())
+        .when(blockAttestationsPool)
+        .getAggregatedAttestationsForBlockAtSlot(any());
+
+    eventBus = spy(new EventBus());
+    storageClient = new ChainStorageClient(eventBus);
+    List<BLSKeyPair> blsKeyPairList =
+        new MockStartValidatorKeyPairFactory().generateKeyPairs(0, numValidators);
+    final BeaconChainUtil chainUtil = BeaconChainUtil.create(storageClient, blsKeyPairList);
     chainUtil.initializeStorage();
 
     vc =
```

Comment:
I think this contradict's Meredith's advice on the PR that first introduced this test (she was right at the time) but now that we need to create different instances of `ValidatorCoordinator` in different tests, I'd move the creation out of the `@BeforeEach`.  It probably is worth having a private method to do the creation so something like:
```
  private ValidatorCoordinator createValidatorCoordinator(final int ownedValidatorCount) {
    when(config.getInteropOwnedValidatorCount()).thenReturn(0);
    return new ValidatorCoordinator(
        eventBus, storageClient, attestationAggregator, blockAttestationsPool, config);
  }
```

Reply:
done.